diff --git a/common/autoconf/generated-configure.sh b/common/autoconf/generated-configure.sh
index d3d0ccd..165429e 100644
--- src/common/autoconf/generated-configure.sh
+++ src/common/autoconf/generated-configure.sh
@@ -16164,8 +16164,8 @@ $as_echo_n "checking flags for boot jdk java command for big workloads... " >&6;
   # Maximum amount of heap memory.
   # Maximum stack size.
   if test "x$BOOT_JDK_BITS" = x32; then
-    JVM_MAX_HEAP=1100M
-    STACK_SIZE=768
+    JVM_MAX_HEAP=512M
+    STACK_SIZE=384
   else
     # Running Javac on a JVM on a 64-bit machine, takes more space since 64-bit
     # pointers are used. Apparently, we need to increase the heap and stack
