Decrease max heap for Raspberry Pi
--- openjdk.orig/common/autoconf/boot-jdk.m4
+++ openjdk/common/autoconf/boot-jdk.m4
@@ -360,11 +360,17 @@ AC_DEFUN_ONCE([BOOTJDK_SETUP_BOOT_JDK_ARGUMENTS],
   # Maximum amount of heap memory.
   # Maximum stack size.
   if test "x$BOOT_JDK_BITS" = x32; then
-    if test "x$OPENJDK_BUILD_CPU_ARCH" = "xs390"; then
-      JVM_MAX_HEAP=768M
-    else
-      JVM_MAX_HEAP=1100M
-    fi
+    case "$OPENJDK_BUILD_CPU_ARCH" in
+      s390)
+        JVM_MAX_HEAP=768M
+        ;;
+      arm)
+        JVM_MAX_HEAP=512M
+        ;;
+      *) 
+        JVM_MAX_HEAP=1100M
+        ;;
+    esac
     STACK_SIZE=768
   else
     # Running Javac on a JVM on a 64-bit machine, takes more space since 64-bit
